File: inappmessaging/app/src/main/java/com/google/firebase/fiamquickstart/EntryChoiceActivity.java
package com.google.firebase.fiamquickstart;

import android.content.Intent;

import com.firebase.example.internal.BaseEntryChoiceActivity;
import com.firebase.example.internal.Choice;

import java.util.Arrays;
import java.util.List;

public class EntryChoiceActivity extends BaseEntryChoiceActivity {

    @Override
    protected List<Choice> getChoices() {
        return Arrays.asList(
                new Choice(
                        "Java",
                        "Run the Firebase In App Messaging quickstart written in Java.",
                        new Intent(this, MainActivity.class)),
                new Choice(
                        "Kotlin",
                        "Run the Firebase In App Messaging quickstart written in Kotlin.",
                        new Intent(this, KotlinMainActivity.class))
        );
    }

}

File: inappmessaging/app/src/main/java/com/google/firebase/fiamquickstart/EntryChoiceActivity.kt
package com.google.firebase.fiamquickstart

import android.content.Intent
import com.firebase.example.internal.BaseEntryChoiceActivity
import com.firebase.example.internal.Choice
import com.google.firebase.fiamquickstart.java.MainActivity
import com.google.firebase.fiamquickstart.kotlin.KotlinMainActivity

class EntryChoiceActivity : BaseEntryChoiceActivity() {

    override fun getChoices(): List<Choice> {
        return listOf(
                Choice(
                        "Java",
                        "Run the Firebase In App Messaging quickstart written in Java.",
                        Intent(this, MainActivity::class.java)),
                Choice(
                        "Kotlin",
                        "Run the Firebase In App Messaging quickstart written in Kotlin.",
                        Intent(this, KotlinMainActivity::class.java))
        )
    }
}
